dwave/cloud/api/client.py

```diff
@@ -131,9 +133,108 @@ def request(self, method: str, *args, **kwargs):
         return response
 
 
-class VersionedAPISession(LoggingSession):
+class PayloadCompressingSession(LoggingSession):
+    """A :class:`requests.Session` subclass (technically, a further specialized
+    :class:`.LoggingSession`) that adds support for payload compression on the
+    fly.
+
+    Args:
+        compress (bool, default=False):
+            Preemptively compress HTTP request body, even on PUT and POST requests.
+            When compression is enabled, the payload is always compressed in 32KiB
+            chunks and streamed to minimize memory overhead.
+
+    .. note::
+        With compression enabled, the request body is transferred using chunked
+        encoding (``Transfer-Encoding: chunked``) and content is compressed using
+        deflate compression algorithm (``Content-Encoding: deflate``).
+
+    .. versionadded:: 0.13.3
+        Preemptive payload compression support added to :class:`.DWaveAPIClient`.
+
+    """
+
+    _compress: bool = False
+
+    def __init__(self, compress: bool = False, **kwargs):
+        self._compress = compress
+        super().__init__(**kwargs)
+
+    def set_payload_compress(self, compress: bool = True) -> bool:
+        """Modify payload compression on upload for current session.
+
+        Args:
+            compress (bool, default=True):
+                Enable request body compression with ``deflate`` on upload
```

Comment:
```suggestion
                Enable request body compression with DEFLATE on upload
```

if you mean here to specify the algorithm. If you mean to say how the header is modified: ``Content-Encoding: deflate``